src/cocaine-app/planner/__init__.py

```diff
@@ -73,12 +78,88 @@ def __init__(self, db, niu, job_processor, namespaces_settings):
+        :param: couples with expired data size > expired_threshold are recommended to be cleaned
+        :return: list of couples with expired records volume above specified
+        """
+        try:
+            # Configure parameters for work with IT
+            yt_cluster = self.params.get('ttl_cleanup', {}).get('yt_cluster', "")
+            yt_token = self.params.get('ttl_cleanup', {}).get('yt_token', "")
+            yt_attempts = self.params.get('ttl_cleanup', {}).get('yt_attempts', 3)
+            yt_delay = self.params.get('ttl_cleanup', {}).get('yt_delay', 10)
+
+            yt_wrapper = YqlWrapper(cluster=yt_cluster, token=yt_token, attempts=yt_attempts, delay=yt_delay)
+
+            aggregation_table = self.params.get('ttl_cleanup', {}).get('aggregation_table', "")
+            base_table = self.params.get('ttl_cleanup', {}).get('tskv_log_table', "")
+            expired_threshold = self.params.get('ttl_cleanup', {}).get('ttl_threshold', 10737418240)
+
+            # XXX: Prepare aggregate table from time to time, not always
+            yt_wrapper.prepare_aggregate_for_yesterday(base_table, aggregation_table)
+
+            couple_list = yt_wrapper.request_expired_stat(aggregation_table, expired_threshold)
+            logger.info("YT request has completed")
+            return couple_list
+        except:
+            logger.exception("Work with YQL failed")
+            return None
+
+    def _ttl_cleanup_planner(self):
+        try:
+            logger.info('Starting ttl cleanup planner')
+
+            with sync_manager.lock(Planner.TTL_CLEANUP_LOCK, blocking=False):
+                self._do_ttl_cleanup()
+
+        except LockFailedError:
+            logger.info('TTl cleanup planner is already running')
+        except Exception as e:
+            logger.exception('Failed to plan ttl cleanup')
+        finally:
+            logger.info('TTL cleanup planner finished')
+            self.__tq.add_task_at(
+                self.TTL_CLEANUP,
+                self.ttl_cleanup_timer.next(),
+                self._ttl_cleanup_planner)
+
+    def _do_ttl_cleanup(self):
+        logger.info('Run ttl cleanup')
+
+        # prechecking for new or pending tasks
+        count = self.job_processor.job_finder.jobs_count(
+            types=jobs.JobTypes.TYPE_TTL_CLEANUP_JOB,
+            statuses=[jobs.Job.STATUS_NOT_APPROVED,
+                      jobs.Job.STATUS_NEW,
+                      jobs.Job.STATUS_EXECUTING,
+                      jobs.Job.STATUS_PENDING])
+
+        if count >= 0:
+            logger.fatal('Found {} unfinished ttl cleanup jobs'.format(count))
+
+        couple_list = self._get_yt_stat()
+        for couple in couple_list:
+            request = {}
+            request['iter_group'] = couple #in tskv coupld id is actually group[0] from couple id
+            request['need_approving'] = not self.params.get('ttl_cleanup', {}).get('autoapprove', False)
+            self.ttl_cleanup(request)
```

Comment:
Done in https://github.com/yandex/mastermind/pull/311/commits/c9fc028033e45de4292205a01c9dc3a012109527